### PR TITLE
Bug 1861647 - Fenix: Add new Addons high priority tests

### PR DESCRIPTION
**Pull request by sv-ohorvath**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1861647

### CODE REVIEW

**Review by AndiAJ — APPROVED** [review]
👍🏽

**Review by jjSDET — APPROVED** [review]
LGTM

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt [hunk 1]
```diff
@@ -116,24 +114,37 @@ class SettingsAddonsTest {
...
-            installAddon(addonName)
-            verifyAddonInstallCompleted(addonName, activityTestRule)
+            installAddon(uBlockAddon)
+            verifyAddonInstallCompleted(uBlockAddon, activityTestRule)
+            closeAddonInstallCompletePrompt()
+            installAddon(darkReaderAddon)
+            verifyAddonInstallCompleted(darkReaderAddon, activityTestRule)
+            closeAddonInstallCompletePrompt()
+            installAddon(privacyBadgerAddon)
+            verifyAddonInstallCompleted(privacyBadgerAddon, activityTestRule)
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt:137]: I'm a little worried for potential network issues here as we're widening the surface.

Looking at 

https://github.com/mozilla-mobile/firefox-android/blob/7fad484fd429a1696de812ab09428f3fe9ab7c49/fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAddonsManagerRobot.kt#L86 - can we add some safety to the install helper? Or does that happen below in `verifyAddonInstallCompleted` ?

Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt]: As discussed with Aaron, I added the verifyAddonInstallCompleted method inside the installAddon() helper to make sure we catch an error while installing the add-on and retry.

**Pull request merged**